devel/libtapi/Portfile

```diff
@@ -19,53 +20,32 @@ long_description        ${description} Newer MacOS SDKs contain only these TAPI
 
 homepage                http://opensource.apple.com/source/tapi/
 
-checksums               rmd160  a99aa279e626d4d638281ef9555d9b5db5f15145 \
-                        sha256  66c715baaea6f28e8c079f36189027a52744138c38e83e97aeb399232365138f \
-                        size    32205809
-
-# use cmake-bootstrap to minimize dependencies.
-depends_build-replace   path:bin/cmake:cmake port:cmake-bootstrap
-configure.cmd           ${prefix}/libexec/cmake-bootstrap/bin/cmake
-depends_skip_archcheck-append cmake-bootstrap
-
-platform darwin {
-    # pick a suitable python to build with
-    if {${os.major} < 11} {
-        # system python is too old, use python27-bootstrap to minimize dependencies
-        depends_build-append    port:python27-bootstrap
-        configure.args-append   -DPYTHON_EXECUTABLE=${prefix}/libexec/python27-bootstrap/bin/python2.7
-        depends_skip_archcheck-append python27-bootstrap
-    } elseif {${os.major} > 19} {
-        # Use system python 3.x
-        configure.args-append   -DPYTHON_EXECUTABLE=/usr/bin/python3
-    } else {
-        # use the system python, which is known to be fine for darwin 11 to 19
-        configure.args-append   -DPYTHON_EXECUTABLE=/usr/bin/python2.7
-    }
-}
+checksums               rmd160  d1ce5001e4ec0452c9a6e2464d72308d0058fdf1 \
+                        sha256  90be4f254bf5df5fc22853536791f533db953c0f772a11f7ea66601fce21d003 \
+                        size    38640440
+
+set python_branch       3.11
+set python_version      [string map {. {}} ${python_branch}]
+
+depends_build-append    port:python${python_version}
+configure.args-append   -DPYTHON_EXECUTABLE=/usr/bin/python${python_branch}
 
 # remove need for port:libxml2 dependency
-configure.args-append -DLIBXML2_LIBRARIES=IGNORE
+configure.args-append   -DLIBXML2_LIBRARIES=IGNORE
 
 # avoid opportunistic linking zlib
-configure.args-append -DLLVM_ENABLE_ZLIB=OFF
+configure.args-append   -DLLVM_ENABLE_ZLIB=OFF
 
 # avoid opportunistic linking ncuses
-configure.args-append -DLLVM_ENABLE_TERMINFO=OFF
+configure.args-append   -DLLVM_ENABLE_TERMINFO=OFF
 
 compiler.cxx_standard   2017
 cmake.build_type        Release
 
-# needs support for llvm::Expect, which requires c++17 inline variables (clang 3.9+)
-compiler.blacklist-append {clang < 801} {macports-clang-3.[3-8]}
-
-# needs gcc8 or later to compile
-# $SRC_DIR/src/llvm/projects/libtapi/lib/Core/APIJSONSerializer.cpp:1202:10: error: could not convert
-# 'result' from 'tapi::internal::API' to 'llvm::Expected<tapi::internal::API>'    return result;
-compiler.blacklist-append *gcc-3.* *gcc-4.* *gcc-5.* *gcc-6.* *gcc-7.*
```

Comment:
Not that I am for blacklisting `gcc7` as long as we have not yet moved to `gcc12` for PowerPC, but does it build with `gcc7`? (I have no idea whether blacklist was needed in the first place, since I have probably built `libtapi` only with `gcc11` and `gcc12`.)

Reply:
@barracuda156 it is suceffuly builded by `gcc7` on my test machine.

Reply:
Great!

Reply:
So it was failing on `ppc` due to size of bool LOL